internal/pkg/gateway/registry.go

```diff
@@ -140,7 +142,7 @@ layout:
 	layouts := append(preferredLayouts, otherLayouts...)
 
 	if len(layouts) == 0 {
-		return nil, fmt.Errorf("failed to select a set of endorsers that satisfy the endorsement policy")
+		return nil, fmt.Errorf("failed to select a set of endorsers that satisfy the endorsement policy due to unavailability of peers: %v", unavailableEndorsers)
```

Comment:
I prefer to use the specific type like %s rather than the generic %v for readability, but that doesn't need to hold up this PR.